### ISSUE TITLE
Error running `doom build` on gccemacs 

### ISSUE THREAD

**Issue by bnjmnt4n** [hlissner/doom-emacs#4303]
**What did you expect to happen?**

I upgraded my gccemacs (pgtk version) to a newer version, and ran doom build, expecting a seamless upgrade.

**What actually happened?**

After running doom build, I get the following error:
```
Debugger entered--Lisp error: (error "Don’t call me!")
  error("Don't call me!")
  define-package("smartparens" "1.11.0" "Automatic insertion, wrapping and paredit-like nav..." ((dash "2.13.0") (cl-lib "0.3")))
  native-elisp-load("/home/bnjmnt4n/.emacs.d/.local/cache/eln/28.0.50-x..." nil)
  #f(compiled-function (process event) #<bytecode -0xdcd8999f5f7a232>)(#<process Compiling: /home/bnjmnt4n/.emacs.d/.local/straight/build-28.0.50/smartparens/smartparens-pkg.el> "finished\n")
  sleep-for(0.1)
  (let ((inhibit-message t)) (sleep-for 0.1))
  (if (/= previous pending) (progn (doom--print (doom--format (format (doom--output-class 'info "\33[KWaiting for %d async jobs...\33[1A" pending)))) (setq previous pending)) (let ((inhibit-message t)) (sleep-for 0.1)))
  (while (progn (setq pending (doom--native-compile-jobs)) (not (= 0 pending))) (if (/= previous pending) (progn (doom--print (doom--format (format (doom--output-class 'info "\33[KWaiting for %d async jobs...\33[1A" pending)))) (setq previous pending)) (let ((inhibit-message t)) (sleep-for 0.1))) (setq --cl-var-- nil))
  (let* ((pending nil) (previous 0) (--cl-var-- t)) (while (progn (setq pending (doom--native-compile-jobs)) (not (= 0 pending))) (if (/= previous pending) (progn (doom--print (doom--format (format (doom--output-class ... "\33[KWaiting for %d async jobs...\33[1A" pending)))) (setq previous pending)) (let ((inhibit-message t)) (sleep-for 0.1))) (setq --cl-var-- nil)) nil)
  doom--wait-for-compile-jobs()
  (progn (doom--compile-site-packages) (doom--wait-for-compile-jobs) (doom--write-missing-eln-errors) (doom--print (doom--format (format (doom--output-class 'success "\33[KRebuilt %d package(s)") (length built)))))
  (if built (progn (doom--compile-site-packages) (doom--wait-for-compile-jobs) (doom--write-missing-eln-errors) (doom--print (doom--format (format (doom--output-class 'success "\33[KRebuilt %d package(s)") (length built))))) (progn (doom--print (doom--format (format (doom--output-class 'success "No packages need rebuilding")))) nil))
  (let* ((built (and t (let* ((recipes recipes) (built nil) (straight-use-package-pre-build-functions (cons ... straight-use-package-pre-build-functions))) (let ((--dolist-tail-- recipes) recipe) (while --dolist-tail-- (setq recipe ...) (let* ... ...) (setq --dolist-tail-- ...)) (setq recipe nil) (nreverse built)))))) (if built (progn (doom--compile-site-packages) (doom--wait-for-compile-jobs) (doom--write-missing-eln-errors) (doom--print (doom--format (format (doom--output-class 'success "\33[KRebuilt %d package(s)") (length built))))) (progn (doom--print (doom--format (format (doom--output-class 'success "No packages need rebuilding")))) nil)))
  (let ((straight-check-for-modifications (if (file-directory-p (straight--modified-dir)) (progn '(find-when-checking)))) (straight--allow-find (and straight-check-for-modifications (executable-find straight-find-executable) t)) (straight--packages-not-to-rebuild (or straight--packages-not-to-rebuild (make-hash-table :test #'equal))) (straight--packages-to-rebuild (or (if force-p :all straight--packages-to-rebuild) (make-hash-table :test #'equal))) (recipes (doom-package-recipe-list))) (add-hook 'comp-async-cu-done-hook #'doom--native-compile-done-h) (if force-p nil (straight--make-build-cache-available)) (let* ((built (and t (let* ((recipes recipes) (built nil) (straight-use-package-pre-build-functions ...)) (let (... recipe) (while --dolist-tail-- ... ... ...) (setq recipe nil) (nreverse built)))))) (if built (progn (doom--compile-site-packages) (doom--wait-for-compile-jobs) (doom--write-missing-eln-errors) (doom--print (doom--format (format (doom--output-class ... "\33[KRebuilt %d package(s)") (length built))))) (progn (doom--print (doom--format (format (doom--output-class ... "No packages need rebuilding")))) nil))))
  (let ((doom-output-indent (+ doom-output-indent-increment doom-output-indent))) (let ((straight-check-for-modifications (if (file-directory-p (straight--modified-dir)) (progn '(find-when-checking)))) (straight--allow-find (and straight-check-for-modifications (executable-find straight-find-executable) t)) (straight--packages-not-to-rebuild (or straight--packages-not-to-rebuild (make-hash-table :test #'equal))) (straight--packages-to-rebuild (or (if force-p :all straight--packages-to-rebuild) (make-hash-table :test #'equal))) (recipes (doom-package-recipe-list))) (add-hook 'comp-async-cu-done-hook #'doom--native-compile-done-h) (if force-p nil (straight--make-build-cache-available)) (let* ((built (and t (let* (... ... ...) (let ... ... ... ...))))) (if built (progn (doom--compile-site-packages) (doom--wait-for-compile-jobs) (doom--write-missing-eln-errors) (doom--print (doom--format (format ... ...)))) (progn (doom--print (doom--format (format ...))) nil)))))
  doom-cli-packages-build(t)
  (if (doom-cli-packages-build (not rebuild-p)) (progn (doom-autoloads-reload)))
  (let ((rebuild-p (cdr (assq 'rebuild-p --alist--)))) (if (doom-cli-packages-build (not rebuild-p)) (progn (doom-autoloads-reload))) t)
  (lambda (--alist--) (ignore --alist--) (let ((rebuild-p (cdr (assq 'rebuild-p --alist--)))) (if (doom-cli-packages-build (not rebuild-p)) (progn (doom-autoloads-reload))) t))(nil)
...
  (unwind-protect (progn (fset #'message vnew) (ignore message) (unwind-protect (catch 'exit (if (and (not (getenv "__DOOMRESTART")) (or doomdir localdir debug-p auto-accept-p)) (progn (if doomdir (progn ... ...)) (if localdir (progn ... ...)) (if debug-p (progn ... ...)) (if auto-accept-p (progn ... ...)) (throw 'exit "__DOOMRESTART=1 $@"))) (delete-file doom-cli-log-file) (delete-file doom-cli-log-error-file) (if help-p (progn (if command (progn ...)) (setq command "help"))) (if (null command) (doom-cli-execute "help") (let ((start-time ...)) (run-hooks 'doom-cli-pre-hook) (if (apply ... command args) (progn ... ...))))) (save-current-buffer (set-buffer log-buffer) (require 'ansi-color) (ansi-color-filter-region (point-min) (point-max))) (let ((dest doom-cli-log-file)) (cond ((bufferp dest) (save-current-buffer (set-buffer dest) (insert-buffer-substring log-buffer))) ((stringp dest) (make-directory (file-name-directory dest) 'parents) (let (... ...) (unwind-protect ... ...)))) (kill-buffer log-buffer)))) (fset #'message old))
  (let* ((vnew #'(lambda (msg &rest args) (save-current-buffer (set-buffer log-buffer) (let (...) (insert ... "\n"))) (if doom-debug-p (doom--print (doom--format ...)) (apply message msg args)))) (old (symbol-function #'message))) (unwind-protect (progn (fset #'message vnew) (ignore message) (unwind-protect (catch 'exit (if (and (not ...) (or doomdir localdir debug-p auto-accept-p)) (progn (if doomdir ...) (if localdir ...) (if debug-p ...) (if auto-accept-p ...) (throw ... "__DOOMRESTART=1 $@"))) (delete-file doom-cli-log-file) (delete-file doom-cli-log-error-file) (if help-p (progn (if command ...) (setq command "help"))) (if (null command) (doom-cli-execute "help") (let (...) (run-hooks ...) (if ... ...)))) (save-current-buffer (set-buffer log-buffer) (require 'ansi-color) (ansi-color-filter-region (point-min) (point-max))) (let ((dest doom-cli-log-file)) (cond ((bufferp dest) (save-current-buffer ... ...)) ((stringp dest) (make-directory ... ...) (let ... ...))) (kill-buffer log-buffer)))) (fset #'message old)))
  (let ((message (symbol-function #'message))) (let* ((vnew #'(lambda (msg &rest args) (save-current-buffer (set-buffer log-buffer) (let ... ...)) (if doom-debug-p (doom--print ...) (apply message msg args)))) (old (symbol-function #'message))) (unwind-protect (progn (fset #'message vnew) (ignore message) (unwind-protect (catch 'exit (if (and ... ...) (progn ... ... ... ... ...)) (delete-file doom-cli-log-file) (delete-file doom-cli-log-error-file) (if help-p (progn ... ...)) (if (null command) (doom-cli-execute "help") (let ... ... ...))) (save-current-buffer (set-buffer log-buffer) (require 'ansi-color) (ansi-color-filter-region (point-min) (point-max))) (let ((dest doom-cli-log-file)) (cond (... ...) (... ... ...)) (kill-buffer log-buffer)))) (fset #'message old))))
  (let* ((log-buffer (generate-new-buffer " *doom log*")) (standard-output #'(lambda (out) (save-current-buffer (set-buffer log-buffer) (insert-char out)) (send-string-to-terminal (char-to-string out))))) (let ((message (symbol-function #'message))) (let* ((vnew #'(lambda (msg &rest args) (save-current-buffer ... ...) (if doom-debug-p ... ...))) (old (symbol-function #'message))) (unwind-protect (progn (fset #'message vnew) (ignore message) (unwind-protect (catch 'exit (if ... ...) (delete-file doom-cli-log-file) (delete-file doom-cli-log-error-file) (if help-p ...) (if ... ... ...)) (save-current-buffer (set-buffer log-buffer) (require ...) (ansi-color-filter-region ... ...)) (let (...) (cond ... ...) (kill-buffer log-buffer)))) (fset #'message old)))))
  (condition-case e (let* ((log-buffer (generate-new-buffer " *doom log*")) (standard-output #'(lambda (out) (save-current-buffer (set-buffer log-buffer) (insert-char out)) (send-string-to-terminal (char-to-string out))))) (let ((message (symbol-function #'message))) (let* ((vnew #'(lambda ... ... ...)) (old (symbol-function #'message))) (unwind-protect (progn (fset #'message vnew) (ignore message) (unwind-protect (catch ... ... ... ... ... ...) (save-current-buffer ... ... ...) (let ... ... ...))) (fset #'message old))))) (doom-cli-command-not-found-error (doom--print (doom--format (format (doom--output-class 'error "Command 'doom %s' not recognized") (string-join (cdr e) " ")))) (doom--print (doom--format (format "\nDid you mean one of these commands?"))) (apply #'doom-cli-execute "help" "--similar" (string-join (cdr e) " ")) 2) (doom-cli-wrong-number-of-arguments-error (let* ((--cl-rest-- (cdr e)) (route (if (= (length --cl-rest--) 5) (car-safe (prog1 --cl-rest-- ...)) (signal 'wrong-number-of-arguments (list ... ...)))) (opt (car-safe (prog1 --cl-rest-- (setq --cl-rest-- ...)))) (arg (car-safe (prog1 --cl-rest-- (setq --cl-rest-- ...)))) (n (car-safe (prog1 --cl-rest-- (setq --cl-rest-- ...)))) (d (car --cl-rest--))) (progn (doom--print (doom--format (format (doom--output-class ... "doom %s: %S requires %d arguments, but %d given\n") (mapconcat ... route " ") arg n d))) (let ((doom-output-indent (+ doom-output-indent-increment doom-output-indent))) (apply #'doom-cli-execute "help" (mapcar #'symbol-name route))))) 3) (doom-cli-unrecognized-option-error (let ((option (car (cdr e)))) (doom--print (doom--format (format (doom--output-class 'error "Unrecognized option: %S") option))) (if (string-match "^--[^=]+=\\(.+\\)$" option) (progn (doom--print (doom--format (format "The %S syntax isn't supported. Use '%s %s' instead..." option ... ...)))))) 4) (doom-cli-deprecated-error (let* ((commands (cdr e)) (route (if commands (car-safe (prog1 commands ...)) (signal 'wrong-number-of-arguments (list ... ...))))) (progn (doom--print (doom--format (format (doom--output-class ... "The 'doom %s' command was removed and replaced wit...") (mapconcat ... route " ")))) (let ((doom-output-indent (+ doom-output-indent-increment doom-output-indent))) (let ((--dolist-tail-- commands)) (while --dolist-tail-- (let ... ... ...)))))) 5) (user-error (doom--print (doom--format (format (doom--output-class 'warn "%s") (car (cdr e))))) 1))
  (let ((help-p (cdr (assq 'help-p --alist--))) (auto-accept-p (cdr (assq 'auto-accept-p --alist--))) (debug-p (cdr (assq 'debug-p --alist--))) (doomdir (cdr (assq 'doomdir --alist--))) (localdir (cdr (assq 'localdir --alist--))) (command (cdr (assq 'command --alist--))) (args (cdr (assq 'args --alist--)))) (condition-case e (let* ((log-buffer (generate-new-buffer " *doom log*")) (standard-output #'(lambda (out) (save-current-buffer ... ...) (send-string-to-terminal ...)))) (let ((message (symbol-function #'message))) (let* ((vnew #'...) (old (symbol-function ...))) (unwind-protect (progn (fset ... vnew) (ignore message) (unwind-protect ... ... ...)) (fset #'message old))))) (doom-cli-command-not-found-error (doom--print (doom--format (format (doom--output-class 'error "Command 'doom %s' not recognized") (string-join (cdr e) " ")))) (doom--print (doom--format (format "\nDid you mean one of these commands?"))) (apply #'doom-cli-execute "help" "--similar" (string-join (cdr e) " ")) 2) (doom-cli-wrong-number-of-arguments-error (let* ((--cl-rest-- (cdr e)) (route (if (= ... 5) (car-safe ...) (signal ... ...))) (opt (car-safe (prog1 --cl-rest-- ...))) (arg (car-safe (prog1 --cl-rest-- ...))) (n (car-safe (prog1 --cl-rest-- ...))) (d (car --cl-rest--))) (progn (doom--print (doom--format (format ... ... arg n d))) (let ((doom-output-indent ...)) (apply #'doom-cli-execute "help" (mapcar ... route))))) 3) (doom-cli-unrecognized-option-error (let ((option (car (cdr e)))) (doom--print (doom--format (format (doom--output-class ... "Unrecognized option: %S") option))) (if (string-match "^--[^=]+=\\(.+\\)$" option) (progn (doom--print (doom--format ...))))) 4) (doom-cli-deprecated-error (let* ((commands (cdr e)) (route (if commands (car-safe ...) (signal ... ...)))) (progn (doom--print (doom--format (format ... ...))) (let ((doom-output-indent ...)) (let (...) (while --dolist-tail-- ...))))) 5) (user-error (doom--print (doom--format (format (doom--output-class 'warn "%s") (car (cdr e))))) 1)))
  (closure ((plist) (aliases) (name . :doom) cl-struct-doom-cli-option-tags cl-struct-doom-cli-tags t) (--alist--) (ignore --alist--) (let ((help-p (cdr (assq 'help-p --alist--))) (auto-accept-p (cdr (assq 'auto-accept-p --alist--))) (debug-p (cdr (assq 'debug-p --alist--))) (doomdir (cdr (assq 'doomdir --alist--))) (localdir (cdr (assq 'localdir --alist--))) (command (cdr (assq 'command --alist--))) (args (cdr (assq 'args --alist--)))) (condition-case e (let* ((log-buffer (generate-new-buffer " *doom log*")) (standard-output #'...)) (let ((message ...)) (let* (... ...) (unwind-protect ... ...)))) (doom-cli-command-not-found-error (doom--print (doom--format (format ... ...))) (doom--print (doom--format (format "\nDid you mean one of these commands?"))) (apply #'doom-cli-execute "help" "--similar" (string-join (cdr e) " ")) 2) (doom-cli-wrong-number-of-arguments-error (let* ((--cl-rest-- ...) (route ...) (opt ...) (arg ...) (n ...) (d ...)) (progn (doom--print ...) (let ... ...))) 3) (doom-cli-unrecognized-option-error (let ((option ...)) (doom--print (doom--format ...)) (if (string-match "^--[^=]+=\\(.+\\)$" option) (progn ...))) 4) (doom-cli-deprecated-error (let* ((commands ...) (route ...)) (progn (doom--print ...) (let ... ...))) 5) (user-error (doom--print (doom--format (format ... ...))) 1))))(((command . "build")))
  funcall((closure ((plist) (aliases) (name . :doom) cl-struct-doom-cli-option-tags cl-struct-doom-cli-tags t) (--alist--) (ignore --alist--) (let ((help-p (cdr (assq 'help-p --alist--))) (auto-accept-p (cdr (assq 'auto-accept-p --alist--))) (debug-p (cdr (assq 'debug-p --alist--))) (doomdir (cdr (assq 'doomdir --alist--))) (localdir (cdr (assq 'localdir --alist--))) (command (cdr (assq 'command --alist--))) (args (cdr (assq 'args --alist--)))) (condition-case e (let* ((log-buffer (generate-new-buffer " *doom log*")) (standard-output #'...)) (let ((message ...)) (let* (... ...) (unwind-protect ... ...)))) (doom-cli-command-not-found-error (doom--print (doom--format (format ... ...))) (doom--print (doom--format (format "\nDid you mean one of these commands?"))) (apply #'doom-cli-execute "help" "--similar" (string-join (cdr e) " ")) 2) (doom-cli-wrong-number-of-arguments-error (let* ((--cl-rest-- ...) (route ...) (opt ...) (arg ...) (n ...) (d ...)) (progn (doom--print ...) (let ... ...))) 3) (doom-cli-unrecognized-option-error (let ((option ...)) (doom--print (doom--format ...)) (if (string-match "^--[^=]+=\\(.+\\)$" option) (progn ...))) 4) (doom-cli-deprecated-error (let* ((commands ...) (route ...)) (progn (doom--print ...) (let ... ...))) 5) (user-error (doom--print (doom--format (format ... ...))) 1)))) ((command . "build")))
  (if cli (funcall (progn (or (progn (and (memq (type-of cli) cl-struct-doom-cli-tags) t)) (signal 'wrong-type-argument (list 'doom-cli cli))) (aref cli 7)) (doom--cli-process cli (remq nil args))) (user-error "Couldn't find any %S command" command))
  (let* ((cli (and t (doom-cli-get command)))) (if cli (funcall (progn (or (progn (and (memq ... cl-struct-doom-cli-tags) t)) (signal 'wrong-type-argument (list 'doom-cli cli))) (aref cli 7)) (doom--cli-process cli (remq nil args))) (user-error "Couldn't find any %S command" command)))
  doom-cli-execute(:doom "build")
  apply(doom-cli-execute :doom "build")
  (catch 'exit (apply #'doom-cli-execute :doom (cdr (member "--" argv))))
  (let* ((val (catch 'exit (apply #'doom-cli-execute :doom (cdr (member "--" argv))))) (pcase-0 #'(lambda (command) (let ((script ...) (coding-system-for-write ...) (coding-system-for-read ...)) (let (... ...) (unwind-protect ... ...)) (set-file-modes script 448)) 128))) (cond ((integerp val) (let ((code val)) code)) ((or (consp val) (or (stringp val) (keywordp val))) (funcall pcase-0 val)) (t 0)))
  (kill-emacs (let* ((val (catch 'exit (apply #'doom-cli-execute :doom (cdr (member "--" argv))))) (pcase-0 #'(lambda (command) (let (... ... ...) (let ... ...) (set-file-modes script 448)) 128))) (cond ((integerp val) (let ((code val)) code)) ((or (consp val) (or (stringp val) (keywordp val))) (funcall pcase-0 val)) (t 0))))
  eval-buffer(#<buffer  *load*> nil "/home/bnjmnt4n/.emacs.d/bin/doom" nil t)  ; Reading at buffer position 6637
  load-with-code-conversion("/home/bnjmnt4n/.emacs.d/bin/doom" "/home/bnjmnt4n/.emacs.d/bin/doom" nil t)
  command-line-1(("-scriptload" "/home/bnjmnt4n/.emacs.d/bin/doom" "--" "build"))
  command-line()
  normal-top-level()
```

**System information:**
<details><pre>
SYSTEM  type       gnu/linux
	config     x86_64-pc-linux-gnu
	shell      /run/current-system/sw/bin/fish
	uname      Linux 5.8.18 #1-NixOS SMP Sun Nov 1 11:45:43 UTC 2020 x86_64
	path       (/nix/store/5jqs9c5xzcxa34ya41lqqi7kd0n4mjs8-emacs-packages-deps/bin /nix/store/a3grbn633nghjkj8im849dzi4nn98706-swaybg-a8f109af90353369e7e2e689efe8ce06eb9c60ac/bin /run/wrappers/bin ~/.nix-profile/bin /etc/profiles/per-user/$USER/bin /nix/var/nix/profiles/default/bin /run/current-system/sw/bin ~/.emacs.d/bin /nix/store/clbzzkdbyhxb0abvyxyyjvbh5kayccsw-emacs-pgtkgcc-20201121.0/libexec/emacs/28.0.50/x86_64-pc-linux-gnu)
EMACS   dir        ~/.emacs.d/
	version    28.0.50
	build      Nov 23, 2020
	buildopts  --prefix=/nix/store/clbzzkdbyhxb0abvyxyyjvbh5kayccsw-emacs-pgtkgcc-20201121.0 --disable-build-details --with-modules --with-x-toolkit=gtk3 --with-cairo --with-nativecomp --with-pgtk CFLAGS=-DMAC_OS_X_VERSION_MAX_ALLOWED=101200
	features   JPEG TIFF GIF PNG RSVG CAIRO SOUND DBUS GSETTINGS GLIB NOTIFY INOTIFY LIBSELINUX GNUTLS LIBXML2 FREETYPE HARFBUZZ LIBOTF ZLIB TOOLKIT_SCROLL_BARS GTK3 PGTK XIM MODULES NATIVE_COMP THREADS LIBSYSTEMD JSON PDUMPER
	traits     (batch server-running envvar-file)
DOOM    dir        ~/.config/doom/
	version    2.0.9
	build      grafted, HEAD -> develop, origin/develop, origin/HEAD c83e5e7 2020-11-21 14:09:08 -0500
	elc-files  0
	modules    (:completion company ivy :ui doom-dashboard hl-todo ligatures modeline ophints (popup +defaults) treemacs vc-gutter vi-tilde-fringe workspaces zen :editor (evil +everywhere) file-templates fold snippets :emacs dired electric ibuffer undo vc :term vterm :checkers syntax spell :tools direnv editorconfig (eval +overlay) lookup lsp magit pdf :lang emacs-lisp json (javascript +lsp) markdown nix (org +pretty +roam) sh web yaml :email notmuch :app (rss +org) :config (default +bindings +smartparens))
	packages   ((modus-operandi-theme) (modus-vivendi-theme) (org-gcal) (org-download) (anki-editor) (mathpix.el :recipe (:host github :repo jethrokuan/mathpix.el)))
	unpin      (n/a)
	elpa       (vterm)
</pre></details>

I'm running emacsPgtkGcc off of emacs-overlay.

**Comment by bnjmnt4n** [comment]
Not sure if this is an error on gccemacs or Doom's side (or an error in my configuration possibly). For now I've reverted to an older version of pgtkgccemacs, and an older version of doom prior to https://github.com/hlissner/doom-emacs/commit/196bb04088bd5370618e3f91c89fc6457d6d9b32. I don't have the bandwidth to debug this for now, but will look into it again possibly next week. Just wanted to put this out there to see if anyone has encountered similar issues.

**Comment by hlissner** [comment]
This is an upstream issue in gccemacs. A regression causes `straight.el` to load each *.el file after compiling it, causing a slew of issues like this one.

It's [been reported upstream to the emacs mailing list](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=44813) and [to straight](https://github.com/raxod502/straight.el/pull/637). 

The only solution to the problem, for now is to revert to an earlier commit of gccemacs.

**Comment by flatwhatson** [comment]
See #4309.

**Comment by bnjmnt4n** [comment]
Thanks for the prompt replies, and for the hard work both of you put in for the emacs community! I'll track the two PRs and attempt an update once they are merged.

**Comment by hlissner** [comment]
This was resolved with #4309 being merged. Let me know if that isn't the case and I'll reopen the issue. Thanks for bringing it to my attention.
